src/Engine/Graphics/DecalBuilder.cpp

```diff
@@ -267,74 +267,71 @@ bool DecalBuilder::ApplyBloodSplatToTerrain(struct Polygon* terrpoly, Vec3f* ter
+                WorldXPosR = triverts[1].vWorldPosition.x;
+                WorldXPosL = triverts[2].vWorldPosition.x;
+                WorldYPosU = triverts[1].vWorldPosition.y;
+                WorldYPosD = triverts->vWorldPosition.y;
             }
+        } else {
+            log->Warning("Uknown strip type detected!");
+        }
 
-            WorldMinZ = pOutdoor->GetPolygonMinZ(triverts, uStripType);
-            WorldMaxZ = pOutdoor->GetPolygonMaxZ(triverts, uStripType);
-
-            // check xy bounds
-            if (WorldXPosR - bloodsplat_container->pBloodsplats_to_apply[i].radius <
-                bloodsplat_container->pBloodsplats_to_apply[i].x &&
-                WorldXPosL + bloodsplat_container->pBloodsplats_to_apply[i].radius >
-                bloodsplat_container->pBloodsplats_to_apply[i].x &&
-                WorldYPosU - bloodsplat_container->pBloodsplats_to_apply[i].radius <
-                bloodsplat_container->pBloodsplats_to_apply[i].y &&
-                WorldYPosD + bloodsplat_container->pBloodsplats_to_apply[i].radius >
-                bloodsplat_container->pBloodsplats_to_apply[i].y) {
-                // check z bounds
-                if (WorldMinZ - bloodsplat_container->pBloodsplats_to_apply[i].radius <
-                    bloodsplat_container->pBloodsplats_to_apply[i].z &&
-                    WorldMaxZ + bloodsplat_container->pBloodsplats_to_apply[i].radius >
-                    bloodsplat_container->pBloodsplats_to_apply[i].z) {
-                    // check plane distance
-                    *tridotdist = -Dot(triverts->vWorldPosition, *terrnorm);
-
-                    planedist = terrnorm->y * bloodsplat_container->pBloodsplats_to_apply[i].y +
-                        terrnorm->z * bloodsplat_container->pBloodsplats_to_apply[i].z +
-                        terrnorm->x * bloodsplat_container->pBloodsplats_to_apply[i].x + *tridotdist;
-                    planedist += 0.5f;
-
-                    if (planedist <= bloodsplat_container->pBloodsplats_to_apply[i].radius) {
-                        // blood splat hits this terrain tri
-
-                        // check if water or something else (maybe should be border tile or swampy>)
-                        if (terrpoly->flags & 2 || terrpoly->flags & 0x100) {
-                            // apply fade flags
-                            if (!(bloodsplat_container->pBloodsplats_to_apply[i].blood_flags & DecalFlagsFade)) {
-                                bloodsplat_container->pBloodsplats_to_apply[i].blood_flags |= DecalFlagsFade;
-                                bloodsplat_container->pBloodsplats_to_apply[i].fade_timer = pEventTimer->Time();
-                            }
+        WorldMinZ = pOutdoor->GetPolygonMinZ(triverts, uStripType);
+        WorldMaxZ = pOutdoor->GetPolygonMaxZ(triverts, uStripType);
+
+        // check xy bounds
+        if (WorldXPosR - bloodsplat_container->pBloodsplats_to_apply[whichsplat].radius <
```

Comment:
`bbox.Expanded(bloodsplat_container->pBloodsplats_to_apply[i].radius).Contains(bloodsplat_container->pBloodsplats_to_apply[i].pos)`

Note that this implies you'll have to change the `struct Bloodsplat`.

Reply:
Having considered further, this check makes no sense anymore. We have a terrain square that gets split into two triangles, passed into this, from which we rebuild a square to test against... Will revise.